behaviour/graql/language/match.feature

```diff
@@ -191,68 +183,897 @@ Feature: Graql Match Clause
+      | FUN |
+
+
+  Scenario: `contains` performs a case-insensitive match
+    Given graql insert
+      """
+      insert
+      $x "The Phantom of the Opera" isa name;
+      $y "Pirates of the Caribbean" isa name;
+      $z "Mr. Bean" isa name;
+      """
+    Given the integrity is validated
+    When get answers of graql query
+      """
+      match $x contains "Bean"; get;
+      """
+    And concept identifiers are
+      |     | check | value                         |
+      | PIR | value | name:Pirates of the Caribbean |
+      | MRB | value | name:Mr. Bean                 |
+    Then uniquely identify answer concepts
+      | x   |
+      | PIR |
+      | MRB |
+
+
+  Scenario: `like` matches strings that match the specified regex
+    Given graql insert
+      """
+      insert
+      $x "ABC123" isa name;
+      $y "123456" isa name;
+      $z "9" isa name;
+      """
+    Given the integrity is validated
+    When get answers of graql query
+      """
+      match $x like "^[0-9]+$"; get;
+      """
+    And concept identifiers are
+      |     | check | value       |
+      | ONE | value | name:123456 |
+      | NIN | value | name:9      |
+    Then uniquely identify answer concepts
+      | x   |
+      | ONE |
+      | NIN |
+
+
+  #######################
+  # ATTRIBUTE OWNERSHIP #
+  #######################
+
+  Scenario: `has` can be used to match things that own any instance of the specified attribute
+    Given graql insert
+      """
+      insert
+      $x isa person, has name "Leila", has ref 0;
+      $y isa person, has ref 1;
+      $c isa company, has name "Grakn", has ref 2;
+      $d isa company, has ref 3;
+      """
+    Given the integrity is validated
+    When get answers of graql query
+      """
+      match $x has name $y; get $x;
+      """
+    And concept identifiers are
+      |     | check | value |
+      | LEI | key   | ref:0 |
+      | GRA | key   | ref:2 |
+    Then uniquely identify answer concepts
+      | x   |
+      | LEI |
+      | GRA |
+
+
+  Scenario: by using the `attribute` meta label, `has` can be used to match things that own any type of attribute with a specified value
+    Given graql define
+      """
+      define
+      shoe-size sub attribute, value long;
+      person has shoe-size;
+      """
+    Given the integrity is validated
+    Given graql insert
+      """
+      insert
+      $x isa person, has age 9, has ref 0;
+      $y isa person, has shoe-size 9, has ref 1;
+      $z isa person, has age 12, has shoe-size 12, has ref 2;
+      """
+    Given the integrity is validated
+    When get answers of graql query
+      """
+      match $x has attribute 9; get;
+      """
+    And concept identifiers are
+      |     | check | value |
+      | AG9 | key   | ref:0 |
+      | SS9 | key   | ref:1 |
+    Then uniquely identify answer concepts
+      | x   |
+      | AG9 |
+      | SS9 |
+
+
+  Scenario: when an attribute instance is fully specified, `has` matches its owners
```

Comment:
can we extend this to match owners from more than 1 type? Eg. add a `Dog` that has a name unrelated to `person` and make sure both dogs and people come back when querying for things with names.

Might even be good to have this working for the other types also owning an attribute: relation and attribute

Reply:
Changed insert query to:
```
insert
      $x isa person, has name "Zoe", has age 21, has graduation-date 2020-06-01, has ref 0;
      $y (friend: $x) isa friendship, has age 21, has ref 1;
      $z 2020-06-01 isa graduation-date, has age 21, has ref 2;
      $w isa person, has ref 3;
      $v (friend: $x, friend: $w) isa friendship, has age 7, has ref 4;
      $u 2019-06-03 isa graduation-date, has age 22, has ref 5;
```